extension/src/experiments/webview/messages.ts

```diff
@@ -216,22 +218,27 @@ export class WebviewMessages {
         )
 
       case MessageFromWebviewType.SHOW_MORE_COMMITS:
-        return this.changeNumberOfCommits()
+        return this.changeCommitsToShow(1)
+
+      case MessageFromWebviewType.SHOW_LESS_COMMITS:
+        return this.changeCommitsToShow(-1)
 
       default:
         Logger.error(`Unexpected message: ${JSON.stringify(message)}`)
     }
   }
 
-  private async changeHasMoreCommits() {
-    this.hasMoreCommits =
-      (await this.getNumCommits()) > this.numberOfCommitsToShow
+  private async changeHasMoreOrLessCommits() {
+    const availableNbCommits = await this.getNumCommits()
+    this.hasMoreCommits = availableNbCommits > this.numberOfCommitsToShow
+    this.isShowingMoreCommits =
+      Math.min(this.numberOfCommitsToShow, availableNbCommits) > 1
   }
 
-  private async changeNumberOfCommits() {
-    this.numberOfCommitsToShow = this.numberOfCommitsToShow + 2
-    await this.getMoreCommits(this.numberOfCommitsToShow)
-    await this.changeHasMoreCommits()
+  private async changeCommitsToShow(change: number) {
+    this.numberOfCommitsToShow = this.numberOfCommitsToShow + 2 * change
```

Comment:
[Q] Can we have a constant instead of `2 *` and plug that into `MessageFromWebviewType.SHOW_MORE_COMMITS` / `MessageFromWebviewType.SHOW_LESS_COMMITS`?